### DISCUSSION TITLE
Can't trigger a rule for part of a string

### DISCUSSION THREAD

**Discussion by Ixiruss** [firefly-iii/firefly-iii#9364] [Firefly III]
Hello,

I'm asking a question here as I'm not sure to check all documentatipn (even if I think so). But is it not possible to do a rule that triggers when part of a specific word is matched ?

For example, I have transactions like : 
ABCbank1234567

I want to trigger a rule : 
Description contains "ABC"

The issue is if "ABC" is not a whole word, it will not be triggered, and I want it to be triggered.

I cannot do a contains with "ABCbank" because there is random numbers without space, and also I have some time "ABC bank123456", so that's why I wan't to set a contains with "ABC" that would catch all scenario...

I don't find the regex that may be useful for this usecase, checked https://docs.firefly-iii.org/how-to/firefly-iii/features/expressions/ and https://symfony.com/doc/current/reference/formats/expression_language.html.

Someone could help me please ?

**Comment by JC5** [comment]
Hi, thanks for starting a discussion. 

The functionality you describe exists already, and works as expected on the demo site.

Is there anything I can help you with?

**Comment by Ixiruss** [comment]
Ok so I investigated further and it is not a contains issue. It is working as intended.
The issue was another rule that searched in one specific account, and I thought my transaction was on the another account that I targeted.
Sorry for this and thanks anyway !

I'm closing the discussion
